folly/Utility.h

```diff
@@ -295,8 +295,11 @@ constexpr auto to_signed(T const& t) -> typename std::make_signed<T>::type {
   // note: static_cast<S>(t) would be more straightforward, but it would also be
   // implementation-defined behavior and that is typically to be avoided; the
   // following code optimized into the same thing, though
+FOLLY_PUSH_WARNING
+FOLLY_MSVC_DISABLE_WARNING(4018)
```

Comment:
Can this be done without suppressing warnings?

```
constexpr auto m = static_cast<T>(std::numeric_limits<S>::max());
return m < t ? ... : ...;
```

Reply:
Yes I like that better.  I initially was avoiding using static_cast due to the comment above, but the constexpr static_cast makes sense.